### ISSUE TITLE
Posts from different post types don't show on the Add Post meta box

### ISSUE THREAD

**Issue by lmartins** [Upstatement/stream-manager#56]
Hi,

First and foremost thank you for this extremely useful plugin.
I've testing out to start using on my WP projects but I've noticed a subtle but significant problem.

Im testing Stream Manager to populate a few post lists on a homepage, where there multiple zones with contents from different post types. In most cases each zone is populated with contents from a single CPT, meaning that I use the provided filters quite a lot:

```php
add_filter('stream-manager/options/id=420', function( $defaults ) {
    $defaults['query'] = array_merge($defaults['query'], array(
		'post_type' => 'page',
		'posts_per_page' => 8,
		'meta_query' => array(
	        array(
	         'key' => '_thumbnail_id',
	         'compare' => 'EXISTS'
	        ),
	    )
	));
    return $defaults;
});
```

In this particular example I'm excluding posts that do not have a featured image, but the issue I'm describing next happens with or without that parameter.

The problem quite simply is that when I try to use the Add Post meta box two things happen:

1) It still suggests posts for this stream, even though the stream is configured to only accept pages
2) It won't suggest any page, no matter exists or not pages that would meet the criteria above.

Anything I can do to avoid this?

**Comment by lggorman** [comment]
Hey @lmartins, thanks for logging this.  Looks like the search suggestions are being powered by a more generic query, whereas the expected behavior would be for them to have the same filter applied to the stream itself, like you described.  So your setup is correct -- the problem is on the plugin side.  I'm taking a look now to see if there's a simple resolution, will keep you updated.

**Comment by lmartins** [comment]
@lggorman thanks so much for the feedback Linda.

**Comment by lggorman** [comment]
@lmartins I believe this should be resolved by #57.  When you get the chance pull the latest (or upgrade from the wp plugin repo) and let me know if the Add Posts search is now working as expected.

**Comment by lmartins** [comment]
@lggorman I think so Linda, seems to be working beautifully now. Thank you for the update :)
